package/upgrade/upgrade_node.sh

```diff
@@ -531,6 +555,9 @@ upgrade_os() {
   # The trap will be only effective from this point to the end of the execution
   trap clean_up_tmp_files EXIT
 
+  # we would like to clean up the incomplete state.yaml to avoid the issue of https://github.com/harvester/harvester/issues/4526
+  cleanup_incomplete_state_file
```

Comment:
suggest moving to a later stage, we can skip it if there is not need to upgrade OS.

Reply:
Updated! thanks for the suggestion